### PR TITLE
feat: products page with filter

### PR DESCRIPTION
**Pull request by ravikumar1002**
* Products listing Page
 * Filter 
     - Sort by
     - Ratings
     - Range
     - Category

https://user-images.githubusercontent.com/86847625/161416507-b3c53a2a-e1a1-4c34-b4a7-f9d27d4d6bef.mp4

### CODE REVIEW

**Review by thegeorgenikhil — APPROVED** [review]
Good use of functions.
Ready to merge.

Suggestions: some functions and event handlers(eg. sort dispatch handler) can be even more abstracted to avoid repeating of code, please keep that in mind while you start refactoring in the end.

**Comment by ravikumar1002** [comment]
> Good use of functions. Ready to merge.
> 
> Suggestions: some functions and event handlers(eg. sort dispatch handler) can be even more abstracted to avoid repeating of code, please keep that in mind while you start refactoring in the end.

yeah sure

File: src/components/filter/Filter.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import "./filter.css";
+import { productsDataContext } from "../../context/products-context/products-data";
+import { useContext } from "react";
+import { RangeFilter } from "./filter-component/Range-filter";
+import { CategoryFilter } from "./filter-component/Category-filter";
+import { RatingsFilter } from "./filter-component/Ratings-filter";
+import { SortFilter } from "./filter-component/Sort-filter";
+const Filter = () => {
```
Comment (manish591) [inline src/components/filter/Filter.jsx:8]: Use barrel imports

**Review by manish591 — APPROVED** [review]
Code is looking good to merge. Code is nicely formatted and naming is good too

**Pull request merged**